TR-4653: Upgrade SQLite.swift to the latest version 0.15.3

SvobodaOcode opened the pull request:
SQLite.swift 0.15.3 from https://github.com/stephencelis/SQLite.swift/releases/tag/0.15.3
+ 1 commit from master https://github.com/stephencelis/SQLite.swift/commit/c7b78f2f9a0376bc0145f3ecbd0df05eab1688f7

Also includes fixes for SQLite.swift build fixes on Windows from
https://github.com/theolampert/SQLite.swift

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/Timing-GmbH/SQLite.swift/7)
<!-- Reviewable:end -->

jberkel committed:
Run on macOS 12

Bump deployment targets

Disable watchOS validation

CocoaPods/CocoaPods#11558

Adjust deployment targets for Xcode 14

geoffmacd committed:
adds support for extended error codes

geoffmacd committed:
lint

marmphco committed:
Fix incorrect column names for SELECT * preceded by a WITH

In  #1139 I introduced support for the `WITH` clause. My implementation contains a bug: the statement preparer doesn't produce the correct result column names for queries containing a `SELECT *` preceded by a `WITH`.

For example, consider the following statement:

```
WITH temp AS ( SELECT id, email from users) SELECT * from temp
```

An error would be thrown when preparing this statement because the glob expansion procedure would try to look up the column names for the result by looking up the column names for the query `SELECT * from temp`. This does not work because `temp` is a temporary view defined in the `WITH` clause.

To fix this, I modified the glob expansion procedure to include the `WITH` clause in the query used to look up the result column names.

marmphco committed:
linter error

Commit:
Merge pull request #1179 from marmphco/bugfix/with-glob

Commit:
Merge pull request #1178 from geoffmacd/extend-error-codes

rex4539 committed:
Fix typos

Commit:
Merge pull request #1182 from rex4539/typos

pongsakorn-onsri committed:
fix Xcode build error

Commit:
Merge pull request #1192 from pongsakorn-onsri/fix/build-error

stefansaasen committed:
Make ithe IndexDefinition properties public

The [documentation](https://github.com/stephencelis/SQLite.swift/blob/master/Documentation/Index.md#querying-the-schema)
suggests that the index schema information can be queried. That requires
the `IndexDefinition` properties to be public.

nathanfallet committed:
Merge pull request #1196 from ohoj-software/bugfix/expose-index-definition-properties

Make ithe IndexDefinition properties public

fwcd committed:
Fix GitHub Actions build badge

Commit:
Merge pull request #1200 from fwcd/fix-build-badge

jberkel committed:
Run on macos-13

jberkel committed:
Automatically download swiftlint / xcbeautifier, lint fixes

jberkel committed:
Fix simulator version

jberkel committed:
Use newer Xcode

jberkel committed:
Fix path

jberkel committed:
sudo

jberkel committed:
Try Xcode 14.2

jberkel committed:
Update docs

jberkel committed:
Use shorthand optional binding

jberkel committed:
Re-enable watch tests

jberkel committed:
Merge pull request #1206 from stephencelis/macos-13

Run CI on macOS 13

jberkel committed:
Handle FK definitions w/o key references

Closes #1199

stefansaasen committed:
Fix column affinity parsing to match how SQLite determines affinity

See https://www.sqlite.org/datatype3.html#determination_of_column_affinity
for how SQLite determines column affinity.

stefansaasen committed:
SchemaReader: return the correct column definition for a composite primary key

The `PRAGMA` `table_info` that is used to return the column definitions,
returns one row for each defined column. The `pk` column contains:

> ... either zero for columns that are not part of the primary key, or the 1-based index of the column within the primary key).

See https://www.sqlite.org/pragma.html#pragma_table_info

Checking whether the `pk` column equals 1 only detects a single primary
key and ignores other columns that are part of a composite primary key.

nathanfallet committed:
Merge pull request #1217 from stefansaasen/bugfix/composite-primary-keys

SchemaReader: return the correct column definition for a composite primary key

JacobHearst committed:
Add optional support for decoding

JacobHearst committed:
Fix copypasta

JacobHearst committed:
Fix missing return

JacobHearst committed:
Fix smoothbrain

nathanfallet committed:
Merge pull request #1224 from JacobHearst/master

Add optional support for decoding

geoffmacd committed:
Implements built-in window functions

geoffmacd committed:
update docs

ivanlares committed:
Add privacy manifest

AnthonyMDev committed:
Add visionOS support to Package.swift

AnthonyMDev committed:
Create visionOS target and test plan

nathanfallet committed:
Using macOS 14 to build

nathanfallet committed:
Xcode 15?

nathanfallet committed:
Testing using iOS 17

Commit:
make fromDatatypeValue throw

When using custom types, sometimes decoding can fail, say due to
changes in the type structure. In this case decoding would fail
and the only way to handle it is forcing a crash.

This change allows you to use `try row.get()` instead. Givng you
the chance to handle the mismatch.

nathanfallet committed:
Merge pull request #1242 from elfredpagan/elfred_pagan_throw_fromDatatypeValue

make fromDatatypeValue throw

nathanfallet committed:
fix: json tests (due to a swift upgrade)

nathanfallet committed:
Merge pull request #1234 from ivanlares/add-privacy-manifest

Add privacy manifest

Commit:
Revert "Add privacy manifest"

nathanfallet committed:
Merge pull request #1243 from stephencelis/revert-1234-add-privacy-manifest

Revert "Add privacy manifest"

nathanfallet committed:
fix: ios deployment target

nathanfallet committed:
Merge pull request #1228 from geoffmacd/window-functions

Implements built-in window functions

nathanfallet committed:
Merge pull request #1218 from stefansaasen/bugfix/fix-column-affinity-parsing

Fix column affinity parsing to match how SQLite determines affinity (Fix #1215)

nathanfallet committed:
Merge pull request #1210 from stephencelis/fix-null-fk-column

Handle FK definitions w/o key references

nathanfallet committed:
Merge branch 'master' into vision-os

nathanfallet committed:
fix: deployment targets?

ivanlares committed:
Add privacy manifest

- fixes build issue https://github.com/stephencelis/SQLite.swift/actions/runs/8019552944/job/21907598856

nathanfallet committed:
Merge pull request #1245 from ivanlares/add-privacy-manifest

Add privacy manifest

nathanfallet committed:
changelogs

ChrisStockbridgeVantaca committed:
Adding failing test that includes an optional struct

This test passes with release 0.14.1, but fails starting with 0.15

ChrisStockbridgeVantaca committed:
Bugfix: returning nil when decoding an optional should not throw an error

steventong committed:
Update CoreFunctions.swift fix typo

fix lower to upper

Commit:
Merge pull request #1249 from steventong/steventong-patch-1

nathanfallet committed:
Merge pull request #1248 from Chris-Stockbridge/master

Fix #1247 support nil case when decoding optionals

SagarSDagdu committed:
Add dependency on custom cocoapods fork

SagarSDagdu committed:
Update deployment targets for Xcode 15

nathanfallet committed:
Merge pull request #1255 from SagarSDagdu/privacy-manifest

Change deployment targets for Xcode 15 and add dependency on custom Cocoapods fork

nathanfallet committed:
Merge branch 'master' into vision-os

nathanfallet committed:
using temp cocoapods (fix xcode 15 build, to be reverted later)

nathanfallet committed:
fixing bundle versions?

nathanfallet committed:
adding source for gems as otherwise it does not resolve correctly in CI

nathanfallet committed:
fix carthage location?

nathanfallet committed:
bump xcode version + carthage visionOS tests

Commit:
Merge pull request #1237 from AnthonyMDev/vision-os

nathanfallet committed:
Changelogs v0.15.1

nathanfallet committed:
fix: visionos to cocoapods

nathanfallet committed:
gonna fix that later

nathanfallet committed:
Merge pull request #1260 from stephencelis/fix/visionos-cocoapods

fix: visionos to cocoapods

SagarSDagdu committed:
Update the marketing version to the version we will be releasing next

For recent releases, the `Info.plist` shipped inside the `xcframeworks` do not match the release version. Fixing that to reflect the marketing version we will be releasing next.

SagarSDagdu committed:
Update `podspec` to include privacy manifest

Update `podspec` to include privacy manifest

nathanfallet committed:
Merge pull request #1265 from SagarSDagdu/privacy-manifest-inclusion-in-podspec

Bugfix: Update `podspec` to include privacy manifest

nathanfallet committed:
v0.15.3 (oups)

dyikai committed:
Add `CustomStringConvertible` for Setter

theolampert committed:
Add standalone flag when building on windows

theolampert committed:
Change condition

theolampert committed:
Vend CSQLite

theolampert committed:
Unify non-apple platforms

theolampert committed:
Remove old trace method

nathanfallet committed:
Merge pull request #1279 from dyikai/master

SvobodaOcode committed:
Fix for Module rename to fix Class SQLiteXXX is implemented in both LinkServices.framework and MyApp error logs

SvobodaOcode committed:
Merge branch 'master' of https://github.com/theolampert/SQLite.swift into theolampert-master

SvobodaOcode committed:
Merge branch 'theolampert-master'

SvobodaOcode committed:
Change to fork of CSQLite which has correct compilation for Windows

SvobodaOcode committed:
Fix dependency from version to branch

SvobodaOcode committed:
Fix merge error: invalid redeclaration of 'fromDatatypeValue'

SvobodaOcode committed:
Merge branch 'master' of github.com:SvobodaOcode/SQLite.swift

SvobodaOcode committed:
Merge upstream/master and resolve conflicts

SvobodaOcode committed:
Fixed iOS & macOS build

MrMage reviewed (changes requested):
Thank you! I did notice some changes to the throwing-ness of some accessor methods, as well as a change in the Expression type of the `like` operator — were those intentionally not taken over?

![Screenshot 2025-02-21 at 09.37.19.png](https://files.reviewableusercontent.io/images/reviewable/117466/oHGudC0uEQ_QRqxZtxY8kMOm/Screenshot%202025-02-21%20at%2009.37.19.png)

![Screenshot 2025-02-21 at 09.38.25.png](https://files.reviewableusercontent.io/images/reviewable/117466/s7x5KRwbCPpb_nWToNWv-1ZU/Screenshot%202025-02-21%20at%2009.38.25.png)

> Reviewed 65 of 65 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Timing-GmbH/SQLite.swift/7#-:-OJbds5sB_7kZiv8xRCh:b-g157a7)* status: all files reviewed, 1 unresolved discussion (waiting on @SvobodaOcode)

___
*[`Package.swift` line 44 at r1](https://reviewable.io/reviews/Timing-GmbH/SQLite.swift/7#-OJbety15MjTMLdnCgQy:-OJbety15MjTMLdnCgQz:bfujb04) ([raw file](https://github.com/Timing-GmbH/SQLite.swift/blob/d759c2c2ea16fe29aff4c0a44a74ce6ba7a7d8a3/Package.swift#L44)):*
> ```Swift
> #if os(Linux) || os(Windows) || os(Android)
> package.dependencies = [
>     .package(url: "https://github.com/SvobodaOcode/CSQLite", branch: "main")
> ```

Should we also fork this and move it to the `Timing-GmbH` repository? (I have prepared https://github.com/Timing-GmbH/CSQLite now.)


<!-- Sent from Reviewable.io -->

MrMage reviewed (commented):
P.S.: We should make sure to merge this with a merge commit rather than "squash and merge" to keep the commit history intact.

> *[Reviewable](https://reviewable.io/reviews/Timing-GmbH/SQLite.swift/7#-:-OJbl1y5-opWaxSP1JqB:b1a9ldx)* status: all files reviewed, 1 unresolved discussion (waiting on @SvobodaOcode)



<!-- Sent from Reviewable.io -->

SvobodaOcode committed:
Changed CSQLite package to Timing-GmbH

SvobodaOcode committed:
fix merge error

SvobodaOcode committed:
reverted missing comment

SvobodaOcode commented:
Thank you.

1. I have fixed the package library, now pointing to CSQLite in Timing-GmbH. https://github.com/Timing-GmbH/SQLite.swift/pull/7/commits/1d9644ea49b25478505b864f02f261709aabcf1f
2. I have fixed two merge errors:
 - in Query.swift there was missing comment `// V.ValueType == V, so the case will succeed` https://github.com/Timing-GmbH/SQLite.swift/pull/7/commits/686a48743a37b72e81cc3b49773af94b727e7273
 - in CoreFunctions.swift there were two functions different `func like()` which returned `Expression<Bool?>` instead of `Expression<Bool>` https://github.com/Timing-GmbH/SQLite.swift/pull/7/commits/118bb610f39d083b834def737b4cba4cf35c65ac

All should be fixed now.

MrMage reviewed (approved):
Thanks! Ah, I now see that the extra `try` calls in the old diff are already present in the new base, that's why they don’t show up in the new diff.

> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/Timing-GmbH/SQLite.swift/7#-:-OJc-Hnr3U-vwSlLoybs:b6nzsmk)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @SvobodaOcode)



<!-- Sent from Reviewable.io -->
